### PR TITLE
Add word breaks on punctuation characters.

### LINKED ISSUE #44346: Moving text caret by "word breaks" moves caret incorrectly
CC @bruvzg 

**Godot version:**
4.0 master 2020-12-14

**OS/device including version:**
WIN 10

**Issue description:**
In text/line edit moving by ctrl+left/right arrow keys skips by words, but does so incorrectly.

In below clips I hold ctrl and press left/right arrow keys until I get to the end of the line.
Godot: Skips symbols (braces, parenthesis, periods, etc)
3jzJOkm2RD

Visual Studio: expected behaviour, stops are parenthesis, periods, etc
aGsNqsHlWL

### PR DESCRIPTION
**Pull request by bruvzg**
Adds word breaks on punctuation to the TextServer word breaking.

Fixes #44346

### CODE REVIEW

File: doc/classes/TextServer.xml [hunk 1]
```diff
@@ -1155,6 +1155,9 @@
 		<constant name="GRAPHEME_IS_SPACE" value="8" enum="GraphemeFlag">
 			Grapheme is whitespace.
 		</constant>
+		<constant name="GRAPHEME_IS_PUNCTUATION" value="16" enum="GraphemeFlag">
```
Comment (akien-mga) [inline doc/classes/TextServer.xml:1158]: The value should be 256 no?

Comment (akien-mga) [reply doc/classes/TextServer.xml:1158]: Unless you reorder them, in which case the later enums should change value too.

Comment (bruvzg) [reply doc/classes/TextServer.xml]: Should be 256, fixed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**